web/simple/app.js

```diff
@@ -98,10 +98,10 @@ ajax('GET', url + '/api/server', function(server) {
     ajax('GET', url + '/api/session?token=' + token, function(user) {
 
         map.getView().setCenter(ol.proj.fromLonLat([
-            user.longitude || server.longitude || 0.0,
-            user.latitude || server.latitude || 0.0
+            parseFloat((window.location.search.match(/[?&]longitude=([^&#]+)/) || [])[1]) || user.longitude || server.longitude || 0.0,
```

Comment:
Can you extract `(window.location.search.match(/[?&]XXX=([^&#]+)/) || [])[1]` into some function? Easier to read and avoid duplication.

Reply:
Makes sense. Done as requested. I also simplified the RegEx a little, as window.location.search may never contain any unescaped # (only window.location.href could), so expanding the group with [^&]+ is enough.